samples/drivers/spi_flash/boards/adafruit_feather_stm32f405.conf

```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2020 Richard Osterloh <richard.osterloh@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_SPI_NOR=y
+
+CONFIG_USB=y
+CONFIG_USB_DEVICE_STACK=y
+CONFIG_USB_DEVICE_PRODUCT="Adafruit STM32F405 Feather"
+CONFIG_USB_DEVICE_VID=0x239A
+CONFIG_USB_DEVICE_PID=0x805A
+CONFIG_USB_UART_CONSOLE=y
```

Comment:
Can you elaborate why USB would be needed for this sample ?

Reply:
It is not strictly needed but with this particular device the console over USB is much easier for debugging.

Reply:
Ok, then this would apply to all samples and then maybe this config could be applied as default board config (or optional config) and removed from this sample ?

Reply:
That is completely up to you. I think it's best to leave it up to the user as they may want to use USB for something else but I personally always use USB CDC console on this board.

Reply:
My point is I don't think that we should modify all samples or tests specifically for this board (that for the 'c' part of the change). Then if configuration is really this is useful, it would be better to centralize it at board level, so it is always available w/o need to add this config to all samples.

Reply:
I totally understand. That would be a separate PR anyway so just tell me if you'd like me to take it out for this PR and I'll update it.

Reply:
Ok, let's focus this PR on SPI flash and have a new PR for this USB config, this would be good indeed.

Reply:
Done